Modules/vc-module-pricing/VirtoCommerce.PricingModule.Data/ExportImport2/PricingFullExportPagedDataSource.cs

```diff
@@ -0,0 +1,134 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using VirtoCommerce.ExportModule.Core.Model;
+
+namespace VirtoCommerce.PricingModule.Data.ExportImport
+{
+    public class PricingFullExportPagedDataSource : IPagedDataSource
+    {
+        private class ExportDataSourceState
+        {
+            public ExportDataSourceState()
+            {
+                Result = Array.Empty<IExportable>();
+            }
+
+            public int TotalCount;
+            public int ReceivedCount;
+            public ExportDataQuery DataQuery;
+            public IEnumerable<IExportable> Result;
+            public Func<ExportDataQuery, IPagedDataSource> DataSourceFactory;
+            public int GetNextTake(int take, int pageSize)
```

Comment:
Had moved pagination calculations.  I think we can't calculate without "nextTake" because we should set "take" variable and can't receive "ReceivedCount" before all tasks done.